src/libutil/filesystem.cpp

```diff
@@ -523,13 +708,19 @@ Filesystem::read_text_file (string_view filename, std::string &str)
 {
     // For info on why this is the fastest method:
     // http://insanecoding.blogspot.com/2011/11/how-to-read-in-file-in-c.html
-    std::ifstream in;
-    Filesystem::open (in, filename);
+    boost::shared_ptr<std::istream> in;
+    {
+        std::istream* inraw;
+        Filesystem::open (&inraw, filename);
+        if (inraw) {
+            in.reset(inraw);
+        }
+    }
+
```

Comment:
Would this section be slightly shorter/clearer as:

    std::istream* inraw;
    Filesystem::open (&inraw, filename);
    boost::shared_ptr<std::istream> in (inraw);

???